### ISSUE TITLE
Automatically transfer No Show hearing tasks

### ISSUE THREAD

**Issue by pshahVA** [department-of-veterans-affairs/caseflow#12539]
### Story
When a hearing No Show task has been open for more than 30 days, the task is automatically completed and the appeal is transferred to the appropriate location.

### Acceptance criteria
- no feature flag needed

Once a No Show hearing task has been open for 30 days:
If the appeal is Legacy, assign it to:
- 55 if the POA / representative type is a Co-located Veteran Service Organization (VSO)
- 81 (Case Storage) if the POA / representative type is any other, including:
  - private agent / attorney
  - state or regional VA organization (e.g. NC Department of Military & Veterans Affairs)
  - no representative
  - a non-co-located Veteran Service Organization (VSO)
- If the appeal is AMA, assign it to Mail (create an EvidenceSubmissionWindow task, which is part of the Mail org)

##### Open questions:
Link to [requirements-gathering doc](https://docs.google.com/document/d/1kcnud1_ZdtVDUujLInp-W8PuZwQafZFZLjbKaOZNSp0/edit?usp=sharing)
- [x] Do the above requirements apply to all or only some Case Types? (Example Case Types: Original, Post Remand, CAVC ...what else?) 
  - [x] Applies to all case types. The important distinction is Legacy vs AMA.
- [x] How can the automatic transfer be reversed? (i.e. what happens when Caseflow automatically transfers a No Show, and it's later discovered that the Veteran did send mail requesting to be rescheduled?) How do we enable a Hearing Coordinator to fetch the Veteran's file so that it's actionable? 
  - [x] The Board staff (attorneys and judges) who review the case to make a decision have the ability to send the case back the Hearing Branch if they determine a hearing needs to be scheduled
- [ ] What is the flow for distributing No Show cases in the Hearings Management queue to individual Coordinators _before the 30-day mark_?
  - [ ] What's the value of this manual step (on demo, there appears to be bulk assign capability)? 
  - [ ] Can Caseflow distribute them to the right Coordinator automatically? 
  - [x] How does an individual Coordinator know when a Veteran has sent in mail (are they alerted; or do they have to proactively look for each specific Legacy Veteran in VACOLS)?
- [x] What is "Mail"? Do we need to create a Mail task, reuse one? What is "Mail" expected to do? What should the language/notification look like for Mail?
  - [x] Evidence Submission Window task for AMA appeals
- [x] Does "case storage" mean that it skips hearings and goes to a decision, or is there another route it takes?
- [x] When does it move to 81 or 55? And what are the exact rules?
- [x] Would it move to location 81 after 15/30 days, or would it move before? Is 30 days appropriate?
- [x] What happens at the 30 day mark?
- [x] What does this mean for an AMA case that is not in VACOLS?
- [x] Who has access to the current "Hearings Management team cases" queue to clear it? 
- [x] How does the transfer of No Show tasks happen currently? What does the queue look like?

### Out of scope
- Automatically route cases to ROs or users

### Release notes
- [] Communicate with the training team that this behavior has changed so they can update their docs

### Designs
n/a

### Background/context

> Currently after a Veteran is scheduled, then no show happens, a task is created and it is sent to the org's “Assigned” queue tab. This is after the Veteran's 15 days to request to have their hearing rescheduled. After the 25 days of hold are up, its hold is removed and it shows up in the organization’s “Unassigned” queue to indicate action is needed. 
> 
> Please refer to this slack thread for more information: 
> https://dsva.slack.com/archives/CJL810329/p1579124576132400
> 
> We move a legacy case to 81 when the hearing tasks are completed: https://github.com/department-of-veterans-affairs/caseflow/blob/98be94ace99a978a8c20249f6177e8aa5b5bbf0b/app/models/tasks/hearing_task.rb#L37-L59

#### Success criteria
Appeals with No Show hearing tasks are automatically moved to the appropriate location after 30 days.

**Comment by tomas-nava** [comment]
waiting on further requirements from hearings branch

**Comment by pshahVA** [comment]
Not sure if this helps: § 20.302 Rule 302. Appeals with a request for a Board hearing.
(a) Except as described in paragraphs (b) and (c) of this section, for appeals in which the appellant requested, on the Notice of Disagreement, a Board hearing, the Board's decision will be based on a review of the following:
(1) Evidence of record at the time of the agency of original jurisdiction's decision on the issue or issues on appeal;
(2) Evidence submitted by the appellant or his or her representative at the hearing, to include testimony provided at the hearing; and
(3) Evidence submitted by the appellant or his or her representative within 90 days following the hearing.
(b) In the event that the hearing request is withdrawn pursuant to § 20.704(e), the Board's decision will be based on a review of evidence described in paragraph (a)(1) of this section, and evidence submitted by the appellant or his or her representative within 90 days following receipt of the withdrawal.
(c) In the event that the appellant does not appear for a scheduled hearing, and the hearing is not rescheduled subject to § 20.704(d), the Board's decision will be based on a review of evidence described in paragraph (a)(1) of this section, and evidence submitted by the appellant or his or her representative within 90 days following the date of the scheduled hearing.
(Authority: 38 U.S.C. 7105, 7107, 7113(b))
[84 FR 182, Jan. 18, 2019]

**Comment by lund5000** [comment]
Completed in https://vajira.max.gov/browse/CASEFLOW-212
